apache/superset
# feat: Customizable email subject name

### SUMMARY
This feature allows users to customize the email subject when creating a report or an alert within Superset. This is driven by a need to include specific parameters within the email subject line, enhancing the specificity and resonance of alert or report notifications. 

Changes:
- Implementation of a new text input field within the 'create alert/report' interface, through which users can specify their desired email subject line.
- If the text field is left blank, a default email subject line will be generated by the system.
- The user interface will show the new text field solely when the email sending method is selected by a user.

This feature has been tested with Superset master branch.
This proposed feature has already been discussed in the Superset Improvement Proposal (see https://github.com/apache/superset/issues/25945)

### BEFORE SCREENSHOTS
![image](https://github.com/apache/superset/assets/76474644/625a349d-0489-4b5e-be79-dbe12a00abac)

![image](https://github.com/apache/superset/assets/76474644/b72808c6-8cf6-44e4-a473-ecb8106e5228)

### AFTER SCREENSHOTS
![image](https://github.com/apache/superset/assets/76474644/8d56feca-cf8e-43e6-9b22-51bec0af484f)

![image](https://github.com/apache/superset/assets/76474644/37747042-30c0-4ace-9d98-e1086ab58e36)

### TESTING INSTRUCTIONS
To verify the changes made, please follow these steps: (Assume that there TAG for alert and report and open)
- Go to the Alert and report creation page
- Create one of them
- Fill all the information that required
- Select Notification method as "Email"
- You will see input box for "email subject name"
- Fill the desired subject name and save it
- Wait for email send and check the email subject name

### ADDITIONAL INFORMATION
